lib/logstash/inputs/file.rb

```diff
@@ -310,4 +315,18 @@ def stop
       @tail.quit
     end
   end
+
+  private
+
+  def exit_flush
+    listener = FlushableListener.new("none", self)
+    if @codec.identity_count.zero?
+      # using the base codec without identity/path info
+      @codec.base_codec.flush do |event|
+        listener.process_event(event) rescue nil
```

Comment:
My thinking here is that this is a final flush done when the pipeline is stopping. An error is unlikely. You are correct though, we should rescue, log and move on.

I will change.